### PR TITLE
feat(atomic): improve atomic-popover interaction

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-1937

Now has
- semi-transparent backdrop which closes on click
- close on ESC key
- focus trap for the button/facet

### CODE REVIEW

File: packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx [hunk 1]
```diff
@@ -211,17 +221,16 @@ export class AtomicPopover implements InitializableComponent {
...
     return (
-      <div class="relative">
-        {this.renderDropdownButton()}
-        <div
-          id={this.popoverId}
-          ref={(el) => (this.popupRef = el!)}
-          part="facet"
-          class={`absolute pt-0.5 z-1 ${this.isOpen ? 'block' : 'hidden'}`}
+      <Host>
+        <atomic-focus-trap
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx:225]: Do I need extra aria attributes here?

Comment (btaillon) [reply packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx:225]: no need for extra aria attributes. However, when the modal opens, the focus gets completely stuck in macOS Voiceover

**Review by btaillon — APPROVED** [review]
make sure this works with all screen readers (as well as the tab key) and make sure to show this to Johnny. Personally, I like the backdrop more than I thought I would.

File: packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx (hunk 1 above)
Comment (ThibodeauJF) [reply packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx:225]: Screen Shot 2022-08-17 at 8 31 20 AM

**Comment by ThibodeauJF** [comment]
I only have voiceover for mac not the other desktop ones & i'll send Johnny the link when it's merged

File: packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx (hunk 1 above)
Comment (ThibodeauJF) [reply packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx:225]: Hmm with the VO - up/down key I cannot even access the button in the first place... But with tab it worked i'll look into it

Comment (ThibodeauJF) [reply packages/atomic/src/components/search/facets/atomic-popover/atomic-popover.tsx:225]: Got it!

**Pull request merged**